sentry_sdk/types.py

```diff
@@ -12,3 +12,5 @@
 
 if TYPE_CHECKING:
     from sentry_sdk._types import Event, Hint  # noqa: F401
+
+    __all__ = ["Event", "Hint"]
```

Comment:
Is it ok that this is behind `TYPE_CHECKING`? Not sure deep into the topic but https://github.com/getsentry/sentry-python/issues/2909 seems to suggest it shouldn't?

Reply:
This PR only fixes #2910. I opened #2914 to fix #2909, and in #2914, I have moved the `__all__` out of the `if TYPE_CHECKING` guard